app/src/parser.c

```diff
@@ -183,11 +183,11 @@ __Z_INLINE parser_error_t parser_formatAmountItem(uint16_t amountToken,
         return parser_unexpected_field;
     }
 
-    if (!parser_areEqual(amountToken + 1u, "amount")) {
+    if (!parser_areEqual(amountToken + 1u, (char*) "amount")) {
```

Comment:
you can change in parser_areEqual the input argument to `const char *expected` and then you don't need to cast the string